src/parsers/sample-models.cpp

```diff
@@ -137,6 +137,207 @@ namespace se3
+     * <pre> is the prefix (string) before every name in the model.
+     */
+    static void addManipulatorGeometries(const Model& model,
+                                         GeometryModel & geom,
+                                         const std::string & pre = "")
+    {
+      GeometryObject shoulderBall(pre+"shoulder_object",
+                                  model.getBodyId(pre+"shoulder1_body"),/*NR*/0,
+                                  boost::shared_ptr<fcl::Sphere>(new fcl::Sphere(0.05)),
+                                  SE3::Identity());
+      geom.addGeometryObject(shoulderBall,model,true);
+      
+      GeometryObject elbowBall(pre+"elbow_object",
+                               model.getBodyId(pre+"elbow_body"),/*NR*/0,
+                               boost::shared_ptr<fcl::Sphere>(new fcl::Sphere(0.05)),
+                               SE3::Identity());
+      geom.addGeometryObject(elbowBall,model,true);
+      
+      GeometryObject wristBall(pre+"wrist_object",
+                               model.getBodyId(pre+"wrist1_body"),/*NR*/0,
+                               boost::shared_ptr<fcl::Sphere>(new fcl::Sphere(0.05)),
+                               SE3::Identity());
+      geom.addGeometryObject(wristBall,model,true);
+
+      GeometryObject upperArm(pre+"upperarm_object",
+                              model.getBodyId(pre+"upperarm_body"),/*NR*/0,
+                              boost::shared_ptr<fcl::Capsule>(new fcl::Capsule(0.05, .8)),
+                              SE3(Eigen::Matrix3d::Identity(), Eigen::Vector3d(0,0,0.5)) );
+      geom.addGeometryObject(upperArm,model,true);
+
+      GeometryObject lowerArm(pre+"lowerarm_object",
+                              model.getBodyId(pre+"lowerarm_body"),/*NR*/0,
+                              boost::shared_ptr<fcl::Capsule>(new fcl::Capsule(0.05, .8)),
+                              SE3(Eigen::Matrix3d::Identity(), Eigen::Vector3d(0,0,0.5)) );
+      geom.addGeometryObject(lowerArm,model,true);
+
+      GeometryObject effectorArm(pre+"effector_object",
+                                 model.getBodyId(pre+"effector_body"),/*NR*/0,
+                                 boost::shared_ptr<fcl::Capsule>(new fcl::Capsule(0.05, .2)),
+                                 SE3(Eigen::Matrix3d::Identity(), Eigen::Vector3d(0,0,0.1)) );
+      geom.addGeometryObject(effectorArm,model,true);
+    }
+
+
+    void manipulator(Model& model) { addManipulator(model); }
+    void manipulatorGeometries(const Model& model, GeometryModel & geom)
+    { addManipulatorGeometries(model,geom); }
+
+    static const Eigen::Vector3d AX_X(1,0,0);
+    static const Eigen::Vector3d AX_Y(0,1,0);
+    static const Eigen::Vector3d AX_Z(0,0,1);
+    
+    static Eigen::Matrix3d rotate(const double angle, const Eigen::Vector3d & axis)
+    { return Eigen::AngleAxisd(angle,axis).toRotationMatrix(); }
+    
+    void humanoid(Model & model, bool usingFF)
```

Comment:
This code is duplicate with respect to buildHumanoidSimple

Reply:
What is duplicated: humanoid function? Yes and no: humanoidSimple is a random kinematic chain with the same tree structure as many humanoids. Here, humanoid is a realistic chain, with realistic masses, no randomization and a set of relevant geometries. I guess that we can mark humanoidSimple as deprecated. 
I have chosen to make a new function rather than overwriting the existing humanoidSimple function, because there is no unittest on top of humanoidSimple, and I am not sure how humanoidSimple is used in Pinocchio or HPP unittest. I did not want to break those tests hence preferred to had something concurrent in the API.

@jmirabel if you are interested by this model for HPP test, you may want to decide some namings of the frames, or add some frames to the tree.

Reply:
humanoidSimple is mostly used for unit tests purposes.

Reply:
Same joint names as in buildHumanoidSimple would be perfect for HPP.

Reply:
Hum, ok, I should not have let my question so open. Your request would be tedious to implement. Could you look at the current model and modify the names as you want. Or even better, add a unittest in pinocchio/unittest/sample-models.cpp about humanoidSimple that is checking the properties that you are expecting from the model. I would then modify the new model to fit your test.

Reply:
I don't think we are expecting any special properties from the model. Joint names are not of uttermost importance as long as they are meaningful and follow a coherent logic.

Reply:
Very good idea !